code/modules/ruins/lavaland_ruin_code.dm

```diff
@@ -178,13 +195,56 @@
 		return
 	if(istype(AM, /mob/living/carbon/human))
 		var/mob/living/carbon/human/H = AM
+		H.influenceSin(ENVY) //Yes, have it sintouch the TARGET,
 		if(user.real_name != H.dna.real_name)
 			user.real_name = H.dna.real_name
 			H.dna.transfer_identity(user, transfer_SE=1)
 			user.updateappearance(mutcolor_update=1)
 			user.domutcheck()
 			user << "You assume the face of [H]. Are you satisfied?"
 
+//Wrath
+
+/obj/item/weapon/melee/wrathhand
+	name = "The hand of hatred."
+	force = 0
+	hitsound = 'sound/magic/WandODeath.ogg'
+	icon_state = "disintegrate"
+	item_state = "disintegrate"
+	flags = NODROP
+	unacidable = 1
+
+/obj/item/weapon/melee/wrathhand/afterattack(atom/movable/AM, mob/living/carbon/human/user, proximity)
+	if(AM == user)
+		user << "<span class='warning'>Surely there is someone you hate more than yourself.</span>"
+	else
+		if(istype(AM,/mob/living))
```

Comment:
You need to check for the proximity flag otherwise it's going to be a ranged gib